### ISSUE TITLE
Inconsistent default spacing rules for JSON operators between PostgreSQL and SQLite

### ISSUE THREAD

**Issue by kgutwin** [sqlfluff/sqlfluff#6404]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

PostgreSQL, MySQL, and SQLite all support the JSON operators `->` and `->>`. However, SQLFluff applies different spacing rules around these operators depending on the dialect.

* For SQLite and MySQL, SQLFluff expects no whitespace around the operator ("touch" behavior).
    * `SELECT a -> 'b'` results in two LT01 failures: "Unexpected whitespace before column path operator '->'." and "Unexpected whitespace before quoted literal."
    * `SELECT a->'b'` results in no linting failures.
* For PostgreSQL, SQLFluff expects whitespace around the operator ("single" behavior).
    * `SELECT a -> 'b'` results in no linting failures.
    * `SELECT a->'b'` results in two LT01 failures: "Expected single whitespace between naked identifier and binary operator '->'." and "Expected single whitespace between binary operator '->' and quoted literal."

This difference in behavior appears to be due to the `postgres` dialect treating the operator as a binary operator, while the `mysql` and `sqlite` dialects are treating it as a "column path operator".

Now personally, I find the "single" style to be more readable in my queries than the "touch" style, but we know that "[If there is one part of building a linter that is going to be controversial it’s going to be whitespace](https://docs.sqlfluff.com/en/stable/configuration/layout.html)". Nevertheless, I tried following the documentation to create a `.sqlfluff` file to get `column_path_operator` to use the "single" style, but it had no effect.

```
[sqlfluff:layout:type:column_path_operator]
spacing_before = single
spacing_within = single
spacing_after = single
```

Therefore, there are probably two things that should be looked at:
1. Consistency in whitespace style around JSON operators between the various dialects
2. Verify that `column_path_operator` or the relevant layout type can have configurable spacing

Many thanks!

### Expected Behaviour

The query `SELECT a -> 'b' FROM foo` should have consistent whitespace lint rules among the `postgres`, `sqlite`, and `mysql` dialects.

### Observed Behaviour

The query `SELECT a -> 'b' FROM foo` fails linting on sqlite and mysql, while the opposite `SELECT a->'b' FROM foo` fails linting on postgres.

### How to reproduce

```bash
cat >test.sql <<EOF
SELECT a -> 'b' FROM foo;
SELECT a->'b' FROM foo;
EOF

# should produce the same output for all three dialects
for dialect in mysql sqlite postgres; do
    sqlfluff lint --verbose --dialect=$dialect test.sql; 
done  
```

### Dialect

sqlite, mysql, postgres

### Version

sqlfluff, version 3.2.5
Python 3.11.10

### Configuration

```
[sqlfluff:layout:type:column_path_operator]
spacing_before = single
spacing_within = single
spacing_after = single
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by rogalski** [comment]
I took a quick look.

MySQL parses JSON operators as special case, which ends up being handled as single `column_reference`:

```
rogal@LAPTOP-6AA568FM MINGW64 ~/repo/sqlfluff (main)
$ echo "SELECT x->'2'" | PYTHONPATH=src python -m sqlfluff parse -d mysql - 
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    column_reference:
[L:  1, P:  8]      |                        naked_identifier:                     'x'
[L:  1, P:  9]      |                        column_path_operator:                 '->'
[L:  1, P: 11]      |                        quoted_literal:                       "'2'"
[L:  1, P: 14]      |                [META] dedent:
[L:  1, P: 14]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    [META] end_of_file:
```

Postgres parses as generic expression:
```
rogal@LAPTOP-6AA568FM MINGW64 ~/repo/sqlfluff (main)
$ echo "SELECT x->'2'" | PYTHONPATH=src python -m sqlfluff parse -d postgres -
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    expression:
[L:  1, P:  8]      |                        column_reference:
[L:  1, P:  8]      |                            naked_identifier:                 'x'
[L:  1, P:  9]      |                        column_path_operator:                 '->'
[L:  1, P: 11]      |                        quoted_literal:                       "'2'"
[L:  1, P: 14]      |                [META] dedent:
[L:  1, P: 14]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    [META] end_of_file:
```

Unfortunately, different vendor docs are not consistent:

- MySQL docs use spaceless snippets: https://dev.mysql.com/doc/refman/8.4/en/json-search-functions.html#operator_json-column-path
- SQLite docs use snippets with spaces: https://www.sqlite.org/json1.html#jptr
- Postgres uses spaces as well: https://www.postgresql.org/docs/current/functions-json.html

I am unsure of how to properly fix, hopefully maintainers have some ideas.
